File: 프로그래머스/1/12947. 하샤드 수/하샤드 수.java
class Solution {
    public boolean solution(int x) {
        
        String str = Integer.toString(x);
        String arr[] = str.split("");
        
        int sum = 0;
        
        for(int i=0; i<arr.length; i++){
            sum += Integer.parseInt(arr[i]);
        }
        
        boolean answer = false;
        if(x%sum == 0) answer = true; 
        
        return answer;
    }
}

File: 프로그래머스/1/12947. 하샤드 수/하샤드 수.js
function solution(x) {
    
    var str = x.toString();
    var arr = str.split("");
    
    var sum = arr.reduce((acc, cur) => acc + Number(cur), 0);

    var answer = false;
    if(x%sum == 0) answer = true;
    
    return answer;
}